src/emm/services/modules_service.py

```diff
@@ -225,9 +232,123 @@ def git_commit_modules(self, commit: str) -> None:
 
         :param commit: Commit content for all changes.
         """
-        enabled_modules = self.get_all_modules(True)
         self.git_service.commit_all(commit)
+        enabled_modules = self.get_all_modules(True)
         for module in enabled_modules:
             module_data = self.get_module_data(module)
             module_location = Path(module_data.prefix) / module
             self.git_service.commit_all(commit, module_location)
+
+    def validate_github_authentication(self) -> bool:
+        """Check if github CLI already authenticated."""
+        return self.github_service.validate_github_authentication()
+
+    def add_pr_links(self, comments: Dict[str, str]) -> Dict[str, List[str]]:
+        """
+        Add link to each module.
+
+        :param comments: Dictionary of module and its pull request url.
+        :return: Dictionary of pull request associate with other modules' pull requests.
+        """
+        pr_dict: Dict[str, List[str]] = defaultdict(list)
```

Comment:
The model for this `pr_dict` seems problematic. It looks like you using a `List` to store the combination of `pr_url` and `pr_links`, then later in the code you are blindly assuming both of those are in your list. Since this appears to be pretty static data, it would be much better to use something like a `NamedTuple` here for a few reasons:

(1) It will avoid mistakes if this code every changes. The code that uses this list is blindly assuming that it will alway get a list of 2 elements. That could be a bad assumption. That code should have checks that ensure there are two elements in the list it gets and do something if that is not the case. With a `NamedTuple`, the extra checks that the data we get looks as expected is unneeded.

(2) It communicates to readers of the code what the data looks like. A `NamedTuple` makes it very clear what data is available, a list tells the reader very little about the data being given, which means it is impossible to understand the `update_pr_links` code without also understanding this function.

Reply:
Thanks for the detailed explain. I was thinking of creating a class as well, but not sure which type to define. NamedTuple is clear and gracefully solves the mapping problem.